src/Npgsql/TypeHandlers/MappedCompositeHandler.cs

```diff
@@ -23,35 +24,26 @@ class MappedCompositeHandler<T> : NpgsqlTypeHandler<T>, IMappedCompositeHandler
     {
         readonly INpgsqlNameTranslator _nameTranslator;
         readonly NpgsqlConnection _conn;
-        [CanBeNull]
-        UnmappedCompositeHandler _wrappedHandler;
+        readonly UnmappedCompositeHandler _wrappedHandler;
 
         public Type CompositeType => typeof(T);
 
-        internal MappedCompositeHandler(INpgsqlNameTranslator nameTranslator, NpgsqlConnection conn)
+        internal MappedCompositeHandler(INpgsqlNameTranslator nameTranslator, PostgresType pgType, NpgsqlConnection conn)
         {
+            PostgresType = pgType;
             _nameTranslator = nameTranslator;
             _conn = conn;
-        }
-
-        void WrapHandler()
-        {
             _wrappedHandler = (UnmappedCompositeHandler)new UnmappedCompositeTypeHandlerFactory(_nameTranslator).Create(PostgresType, _conn);
+            Debug.Assert(_wrappedHandler != null);
```

Comment:
Please remove, there's no way this assert can ever fail one line after setting `_wrappedHandler`